test/tests/functional/pbs_array_job_mail.py

```diff
@@ -76,18 +76,19 @@ def test_emails(self):
                   ("PBS Job Id: " + subjob_jid, "Begun execution"),
                   ("PBS Job Id: " + subjob_jid, "Execution terminated")]
 
-        self.logger.info("Wait 10s for saving the e-mails")
-        time.sleep(10)
-
-        ret = self.du.cat(filename=mailfile, sudo=True)
-        maillog = [x.strip() for x in ret['out'][-600:]]
-
         for (jobid, msg) in emails:
             emailpass = 0
-            for i in range(0, len(maillog)-2):
-                if jobid == maillog[i] and msg == maillog[i+2]:
-                    emailpass = 1
-
+            for j in range(5):
+                time.sleep(5)
+                ret = self.du.cat(filename=mailfile, sudo=True)
+                maillog = [x.strip() for x in ret['out'][-600:]]
```

Comment:
If you only want the last 600 lines, maybe tail is a better option (like you use below).